api/opds.py

```diff
@@ -1381,60 +1422,60 @@ def active_loans_for(cls, circulation, patron, test_mode=False):
 
         feed_obj = AcquisitionFeed(db, "Active loans and holds", url, works, annotator)
         annotator.annotate_feed(feed_obj, None)
-        return feed_obj
+        return feed_obj.as_response(max_age=60*30, private=True)
 
     @classmethod
-    def single_loan_feed(cls, circulation, loan, test_mode=False):
-        db = Session.object_session(loan)
-        work = loan.license_pool.work or loan.license_pool.presentation_edition.work
-        annotator = cls(circulation, None, loan.library,
-                        active_loans_by_work={work:loan},
-                        active_holds_by_work={},
-                        test_mode=test_mode)
-        identifier = loan.license_pool.identifier
-        url = annotator.url_for(
-            'loan_or_hold_detail',
-            identifier_type=identifier.type,
-            identifier=identifier.identifier,
-            library_short_name=loan.library.short_name,
-            _external=True
-        )
-        if not work:
-            return AcquisitionFeed(
-                db, "Active loan for unknown work", url, [], annotator)
-        return AcquisitionFeed.single_entry(db, work, annotator)
+    def single_item_feed(cls, circulation, item, fulfillment=None, test_mode=False,
+                         feed_class=AcquisitionFeed, **response_kwargs):
```

Comment:
Previously, when generating a feed for a specific FulfillmentInfo, the FulfillmentInfo was passed in as `item`. Now the FulfillmentInfo is the optional `fulfillment` argument, and `item` is always a loan or hold. I think this makes more sense because it makes the signature to this method look more like the signature for `acquisition_links`, which is where the loan/hold/fulfillment end up. And in fact, in `test_acquisition_links`, we do test a case where both `loan` and `fulfillment` is being passed in, which previously never happened.